api/services/Flairs.js

```diff
@@ -317,7 +317,7 @@ exports.refreshAppClaim = function (ref, mod_username) {
   } else if (References.isInvolvement(ref) || References.isGiveaway(ref) && /reddit\.com\/r\/pokemontrades/.test(ref.url)) {
     query = {sub: 'pokemontrades', flair: 'involvement'};
   } else if (References.isEgg(ref)) {
-    query = {sub: 'svexchange', flair: {not: {endsWith: 'ribbon'}}};
+    query = {sub: 'svexchange', flair: {$not: /ribbon$/}};
```

Comment:
So apparently Waterline's `not` operator is equivalent to 'not equal', which means that it can only compare to literal values. Mongo's `$not` operator actually inverts the query that follows, which is what we want.